sample-app/src/components/SearchBar.tsx

```diff
@@ -19,17 +20,20 @@ interface Props {
  */
 export default function SearchBar({ placeholder, isVertical, screenReaderInstructionsId }: Props) {
   const answersActions = useAnswersActions();
-  const query = useAnswersState(state => state.query.query);
-  const mapStateToAutocompleteResults: StateSelector<AutocompleteResult[] | undefined> = isVertical
-    ? state => state.vertical.autoComplete?.results
-    : state => state.universal.autoComplete?.results;
-  const autocompleteResults = useAnswersState(mapStateToAutocompleteResults) || [];
-  const isLoading = useAnswersState(state => state.vertical.searchLoading || state.universal.searchLoading);
-
-  function executeAutocomplete () {
+  const query = useAnswersState(state => state.query.input);
+  const [ autocompleteResults, setAutoCompleteResults ] = useState<AutocompleteResult[]>([]);
+  const isLoading = useAnswersState(state => state.searchStatus.isLoading);
+  const networkIds = useRef({ request: 0, response: 0 });
+  async function executeAutocomplete () {
+    const requestId = ++networkIds.current.request;
+    let response = null;
     isVertical
-      ? answersActions.executeVerticalAutoComplete()
-      : answersActions.executeUniversalAutoComplete()
+      ? response = await answersActions.executeVerticalAutocomplete()
+      : response = await answersActions.executeUniversalAutocomplete();
+    if (requestId >= networkIds.current.response) {
+      setAutoCompleteResults(response?.results || []);
+      networkIds.current.response++;
```

Comment:
Do we always increment this? Or should we set `networkIds.current.response` to the `requestid`?

Reply:
ah yeah, should be to requestId!